code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -122,25 +126,29 @@
 			update_inv_glasses()
 		if(ITEM_SLOT_GLOVES)
 			if (gloves && swap)
-				gloves.dropped(src, TRUE)
 				current_equip = gloves
+				if (!temporarilyRemoveItemFromInventory(gloves))
+					return
 			gloves = I
 			update_inv_gloves()
 		if(ITEM_SLOT_FEET)
 			if (shoes && swap)
-				shoes.dropped(src, TRUE)
 				current_equip = shoes
+				if (!temporarilyRemoveItemFromInventory(shoes))
+					return
 			shoes = I
 			update_inv_shoes()
 		if(ITEM_SLOT_OCLOTHING)
-			if (wear_suit && swap)
-				wear_suit.dropped(src, TRUE)
+			var/obj/item/s_store_backup = s_store
+
+			if (swap && wear_suit)
 				current_equip = wear_suit
+				if (!temporarilyRemoveItemFromInventory(wear_suit))
+					return
 
 			wear_suit = I
 
-			if (s_store && swap)
-				var/obj/item/s_store_backup = s_store
+			if (swap && s_store_backup)
```

Comment:
I assume this is a cleanliness thing, isn't it a bit odd to do the typechange a good bit before you do any work on it?

Reply:
It's not a type change, it's to store the old suit storage. This is because changing the suit drops the suit storage item, and sets it to null.